### DISCUSSION TITLE
Tailwind v4 Oxide

### DISCUSSION THREAD

**Discussion by lil5** [saadeghi/daisyui#2899] [General]
With the news on tailwind v4 https://tailwindcss.com/blog/tailwindcss-v4-alpha
How will this impact daisy ui?
Will you still need postcss?

**Comment by saadeghi** [comment]
Post CSS is not a dependency of daisyUI anyway.

There current alpha release doesn't support plugins yet (as mentioned in the blog post) so we don't know yet how's the API going to be.

According to the blog post, it will still be possible to use plugins in using the JS config file as before (in the stable version)

And according to Adam's Twitter, it will also be possible to use plugins directly in the CSS file. The syntax probably will be `@import "daisyui"`

And hopefully (we don't know yet) it will be possible to import CSS files instead of CSS in JS. If so, it will be a big favor for daisyUI because currently the source files are CSS but we have to convert them to CSS in JS to be able to be imported to Tailwind because the current API only accepts CSS in JS.

There's no documents or news about how the API for plugins is going to be so we don't know yet. And in the current alpha release plugins are not supported.

I'm excited about Tailwind 4 and moving forward to use the CSS approach and it will also be the same direction daisyUI is going to move forward.

Let me know if you have any questions

**Reply by ajaypatelaj** [reply]
> The syntax probably will be @import "daisyui"
🤞, I would be great if works like that

> Hopefully (we don't know yet) it will be possible to import CSS files instead of CSS in JS.

That's correct, as per Adam's [post](https://tailwindcss.com/blog/tailwindcss-v4-alpha) the first point of his roadmap is to support `tailwind.config.js`. If this comes true, we can still use the DasiyUI existing plugin mechanism, Right?
_The reason behind asking this question is that we've already started our project with DaiyUI and would like to be optimistic for DaisyUI to continue using it._

![image](https://github.com/saadeghi/daisyui/assets/749684/29685387-87ec-434b-9b10-ff8a66107263)


>And hopefully (we don't know yet) it will be possible to import CSS files instead of CSS in JS. If so, it will be a big favor for daisyUI because currently the source files are CSS but we have to convert them to CSS in JS to be able to be imported to Tailwind because the current API only accepts CSS in JS.
Right, currently DasiyUI creates CSS first and converts them to JS files so that can be used as the TailwindCSS plugin.  

As per Adams's vision for v4 [CSS First Configuration](https://tailwindcss.com/blog/tailwindcss-v4-alpha#css-first-configuration), Let's assume that TailwindCSS v4 provides an API to use Plugin as CSS, will be provided support for that? 
How that would be possible? (I can understand that it's too early a question for now)
What would be your planning and vision for the same for DaisyUI?

Apologies for being a bit insistent, but we strongly prefer to continue utilizing DaisyUI. However, it would greatly benefit both the DaisyUI community and yourself if you could offer ongoing support for future updates and ensure compatibility for those integrating it into their existing projects.

**Reply by saadeghi** [reply]
>Let's assume that TailwindCSS v4 provides an API to use Plugin as CSS, will be provided support for that?

Yes, of course.

>How that would be possible? 

Currently I have as much information as anyone else. They didn't made the API or V4 docs public yet so we don't know how it's going to work. Whatever it is going to be, daisyUI will go with the same direction as Tailwind CSS.

>What would be your planning and vision for the same for DaisyUI?

Yes, of course.

**Comment by oneezy** [comment]
i have tw v4 example setup so daisy ui devs can start porting over (very easy!) 

i already commented on another discussion so i'll just link it here:
https://github.com/saadeghi/daisyui/discussions/2894#discussioncomment-9631271

**Comment by Frikadellios** [comment]
Hi all 👋 

    I am find funny thing)

When you use tailwind.config.(m)js file with tw4 and upload plugins (daisy for example) you need to use tailwind 4 CLI for update configuration ... And then you have already done .css config file with pre configuration for all browsers if you using postcss...(And browserlistrc >2%) 

So, then tou need optimize your code for production, use lightningcss for minify (with lightningcss CLI) and use it 😎 

I do this with 2-3 libraries of Tailwind UI (shadcn for example), and clear code (borders config and other) working well 

Try to do this with Daisy, and create new release ☺️🐈❤️

**Reply by PruthviPraj00** [reply]
Hello @Frikadellios,

Regarding the statement from [TailwindCSS V4.0](https://tailwindcss.com/blog/tailwindcss-v4-alpha):

_**Plugins and custom utilities** — we don’t have support for plugins, or for writing custom utilities that automatically work with variants yet. Obviously, we will make this work before a stable release._

The way I understand It suggests that TailwindCSS V4.0 currently doesn't support plugins like DaisyUI. Could you explain how you were able to use DaisyUI with Tailwind V4.0?

Could you please help me configure daisyUI within TailwindCSS V4.0?

**Reply by Frikadellios** [reply]
I don't test it yet with Daisy UI (try it today later and give you feedback)..

I mean, when you create project with bun:

`bun init --yes`

You need to add @types/node and latest typescript first of all... 

After this add vite@latest or astro@latest in my case.. 

Next step - create standard pages and add utilities (without auto integration like bunx astro add react or npx astro add react) and then you need to install tailwind 4 with all steps (vite plugin, postcss and tailwind command line) and add tailwind.config.js with daisy UI plugin in configuration... And when you use tailwind 4 CLI for you *css main file, all configuration will be uploaded to output file. So, I think, it's working a same with Daisy UI. 

I do this with Ada Html (Tailus UI free template) 2 day's ago, and receive all configurations in one file for Tailwind 4, use lightningcss CLI (with bun and vite plugin) and got ready styled system for tailwind 4.. 

I am leaving in Ukraine, and don't have electricity now 😂, so can show you later a repo from idx.google.com

**Comment by saadeghi** [comment]
I started this discussion for daisyUI 5 alpha: 
https://github.com/saadeghi/daisyui/discussions/3246

Let me if you have any feedbacks, suggestions, questions
